### ISSUE TITLE
Relation entities are not flushed to the database

### ISSUE THREAD

**Issue by ikovac** [mikro-orm/mikro-orm#2161]
I have a problem that relation entities are not flushed to the database.
I have 3 entities: `User`, `Certification`, and `Assignment`.

*user.model.ts*
```js
@Entity()
class User {
  @PrimaryKey()
  id: number;

  @Property()
  name: string;

  @Enum()
  role: Role;

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date();

  @OneToMany(() => Assignment, assignment => assignment.user)
  assignments = new Collection<Assignment>(this);

  constructor(name: string, role: Role) {
    this.name = name;
    this.role = role;
  }
}
```

*certification.model.ts*
```js
@Entity()
class Certification {
  @PrimaryKey()
  id: number;

  @Property()
  name: string;

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date();

  @OneToMany(() => Assignment, assignment => assignment.certification)
  assignments = new Collection<Assignment>(this);

  constructor(name) {
    this.name = name;
  }
}
```

*assignment.model.ts*
```js
@Entity()
class Assignment {
  @Enum({ nullable: true })
  assignmentType?: AssignmentType;

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date();

  @ManyToOne(() => User, { primary: true })
  user!: User;

  @ManyToOne(() => Certification, { primary: true })
  certification!: Certification;

  constructor(certification: Certification, user: User, assignmentType: AssignmentType) {
    this.certification = certification;
    this.user = user;
    this.assignmentType = assignmentType;
  }
}
```

✅ The following code works as expected and all entities are flushed to the database:
```js
const certification = new Certification('first certification');
const certRepo = provider.em.getRepository(Certification);

const user = new User('John Doe', 'user');

const assignment = new Assignment(certification, user, 'primary_employee');

certification.assignments.add(assignment);
await certRepo.persistAndFlush(certification);
```

❌ But, If I persist the certification entity after creating a new certificate, relation entities are not flushed to the database (only certification entity is saved to the database):

```js
const certification = new Certification('first certification');
const certRepo = provider.em.getRepository(Certification);
certRepo.persist(certification);

const user = new User('John Doe', 'user');

const assignment = new Assignment(certification, user, 'primary_employee');

certification.assignments.add(assignment);
await certRepo.flush();
```

✅ If I put another `flush` at the end it works:
```js
const certification = new Certification('first certification');
const certRepo = provider.em.getRepository(Certification);
certRepo.persist(certification);

const user = new User('John Doe', 'user');

const assignment = new Assignment(certification, user, 'primary_employee');

certification.assignments.add(assignment);
await certRepo.flush();
await certRepo.flush();
```

I am confused why do I need to call another `certRepo.flush` method for this to work as expected?
Thanks in advance for any help!

**Versions**
| Dependency | Version |
| - | - |
| node | 16.1.0 |
| typescript | 4.4.2  |
| mikro-orm | 4.5.9 |
| postgresql | 4.5.9 |

**Comment by B4nan** [comment]
Interesting, that will be because when you call persist, it will be cascaded at that time - but there are not relations yet. Then at the flush call, even if we would re-persist, it will see the entity as already persisted (part of persist stack) and just ignore it. 

> I am confused why do I need to call another certRepo.flush method for this to work as expected?

First flush call will clear the persist stack, and add the entity to identity map. Second call will go thru identity map and persist things again.

**Comment by B4nan** [comment]
Should be fixed in `4.5.10-dev.0`

**Comment by ikovac** [comment]
Yeah, it works now 🎉. Thanks for the fast response and fix!
